cmd/dra-example-kubeletplugin/state.go

```diff
@@ -173,3 +203,81 @@ func (s *DeviceState) prepareDevices(claim *resourceapi.ResourceClaim) (Prepared
 func (s *DeviceState) unprepareDevices(claimUID string, devices PreparedDevices) error {
 	return nil
 }
+
+func (pds PreparedDevices) GetDevices() []*drapbv1.Device {
+	var devices []*drapbv1.Device
+	for _, pd := range pds {
+		devices = append(devices, &pd.Device)
+	}
+	return devices
+}
+
+type ptrImplementsRuntimeObject[T any] interface {
+	*T
+	runtime.Object
+	GroupVersionKind() schema.GroupVersionKind
+}
+
+// GetMergedOpaqueDeviceConfig merges all opaque configs specified for a device request in a resource claim.
+//
+// Such configs can either be attached to the resource claim itself or to the
+// device class associated with the request. Configs found directly in the
+// resource claim take precedence over configs found in the device class.
+// Moreover, configs found earlier in the list of configs attached to either a
+// resource claim or a device class take precedence over configs found later in
+// their corresponding list.
+//
+// When merging, a config with higher precedence will overwrite fields of a
+// config with lower precedence. The only exception being that zero-valued
+// fields of higher precedence configs will *not* overwrite non-zero valued
+// fields of lower precedence configs.
+//
+// As such, the final config returned by this function will contain the merged
+// values of all non-zero fields across each of the possible configs in
+// precedence order.
+func GetMergedOpaqueDeviceConfig[T any, TI ptrImplementsRuntimeObject[T]](
+	decoder runtime.Decoder,
+	driverName string,
+	request string,
+	resourceClaim *resourceapi.ResourceClaim,
+	deviceClasses map[string]*resourceapi.DeviceClass,
+) (T, error) {
+	// Collect all configs in order of reverse precedence. Device class configs
+	// have lower precedence than resource claim configs. Within either a
+	// device class or a resource claim, configs earlier in the list have lower
+	// precedence than configs later in the list.
```

Comment:
Given the docstring above, I think we can just use:
```suggestion
	// Collect all configs in order of reverse precedence.
```

Reply:
Updated.